pkg/api/datasources.go

```diff
@@ -274,10 +272,13 @@ func AddDataSource(c *models.ReqContext) response.Response {
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
+
 	datasourcesLogger.Debug("Received command to add data source", "url", cmd.Url)
 	cmd.OrgId = c.OrgId
-	if resp := validateURL(cmd.Type, cmd.Url); resp != nil {
-		return resp
+	if cmd.Url != "" {
```

Comment:
This feels like validation won't happen if URL is empty?

Reply:
Exactly. When you create a new data source using the Grafana UI, the `AddDataSource()` function is called with an empty URL, just basic info about the new data source. The UI then redirects you to the `/edit` page, where you need to specify the URL.

Reply:
Hmm yeah understood, but what about those integrating via the API?

Reply:
The validation is currently being skipped for empty URL strings. The only thing that changed is that I moved that from the `validateURL()` function to the handler to ensure URL validation at update time. If someone is using the API then they can already send empty URL strings to create a data source, and this PR won't change that.

Reply:
All good then :+1: